UI/SidePanel.cpp

```diff
@@ -1287,10 +1286,12 @@ std::shared_ptr<const Ship> ValidSelectedColonyShip(int system_id) {
         return nullptr;
 
     // is there a valid selected ship in the active FleetWnd?
-    for (int ship_id : FleetUIManager::GetFleetUIManager().SelectedShipIDs())
-        if (auto ship = GetShip(ship_id))
-            if (ship->SystemID() == system_id && ship->CanColonize() && ship->OwnedBy(HumanClientApp::GetApp()->EmpireID()))
-                return ship;
+    for (const auto& ship : Objects().find<Ship>(FleetUIManager::GetFleetUIManager().SelectedShipIDs())) {
```

Comment:
This is probably fine in practice, but might warrant a bit more consideration, since getting all the selected ship IDs is not in general necessary, since the first found valid selected colony ship is returned.

Reply:
Now with having the ObjectMap in place for most interfaces it would be feasible to check where filter or transformation ranges could be applied.